### PR TITLE
Propagate Alectryon error codes

### LINKED ISSUE #60: `make pdf` can succeed with failed snippets.
I'm currently trying to build the PDF document with a Nix-installed Alectryon and SerAPI and I'm facing some issues which lead to external Coq modules not being found. This has led me to discover some issues when some snippet evaluation fails.

On page 194 of the PDF, I get the unexpected output:

```
The reference naryFunc was not found in the current environment.
```

But still the PDF was built successfully. This means that errors in PDF generation could go undetected.

In this case, the snippet is `movies/snippets/AckNotPR/majorizedDefs`. Running `make AckNotPR.tex` in `doc/movies` gives:

```
Convert '../../theories/ordinals/MoreAck/AckNotPR.v' to latex.
<string>:(11:4)-(11:68): (WARNING/2) Coq raised an exception:
     > Cannot load Equations.Init: no physical path bound to Equations
     >
   The offending chunk is delimited by >>>…<<< below:
     >
     >
     > >>>Require Import primRec Arith ArithRing List Ack MoreVectors Lia.<<<
     > Require Import Compare_dec Max.
     > Import extEqualNat  VectorNotations. 
   Results past this point may be unreliable.
```

and a lot more similar warnings but in the end the command exits with success.

This comes directly from the behavior of the underlying `python3 extract_snippets.py AckNotPR.tex -o ./snippets/ -d -R ../../theories/ordinals hydras` command.

And this probably comes in turn from Alectryon not raising an exception in such cases. I don't know if there is a way to make Alectryon raise an exception instead of printing a warning (cc @cpitclaudel).

To reproduce, I guess you just need to uninstall Equations before running `make pdf` (but after having built all the `vo` files with `make`).

### PR DESCRIPTION
**Pull request by cpitclaudel**
This should fix #60. @Zimmi48 , could you have a look? It requires the latest Alectryon; I can make a release if needed.

### CODE REVIEW

File: theories/ordinals/Prelude/Iterates.v [hunk 1]
```diff
@@ -54,7 +54,7 @@ Qed.
 (** TODO : move to more generic libraries *)
 
 Lemma iterate_compat3 f x n :
-  Iterates.iterate f n x = nat_rec (fun _ : nat => nat) x (fun _ y : nat => f y) n.
+  iterate f n x = nat_rec (fun _ : nat => nat) x (fun _ y : nat => f y) n.
```
[the remark this answers is not in the document]
Comment (Zimmi48) [reply theories/ordinals/Prelude/Iterates.v]: I tried updating again (to https://github.com/cpitclaudel/alectryon/commit/b3ccc51088e2546a37ff11dc61cd4dca7b9ce2a5) and removing this patch, but this fails again. Any idea why?

[the remark this answers is not in the document]
Comment (cpitclaudel) [reply theories/ordinals/Prelude/Iterates.v]: Hm, no, I'm not 100% sure.  It works in the test I have for it (`corner_cases.rst`), so I will need to investigate further

[the remark this answers is not in the document]
Comment (cpitclaudel) [reply theories/ordinals/Prelude/Iterates.v]: It works for me the new driver.  With `extract_snippets.py`, the problem is a missing `source_path` in the call to `publish_string(` (and now to `_gen_docutils`)

File: doc/movies/extract_snippets/docutils.py [hunk 2]
```diff
@@ -37,17 +39,15 @@ def coq_rst_to_latex(source: str) -> str:
...
-    parser = Parser()
-    return publish_string(
-        source=source.encode("utf-8"),
-        parser=parser, parser_name=None,
-        writer=LuaLatexWriter(), writer_name=None,
-        settings=None, settings_spec=None,
-        settings_overrides=settings_overrides, config_section=None,
-        enable_exit_status=True).decode("utf-8")
+    output, _pub, exit_code = \
+        _gen_docutils(source, None,
```
Comment (cpitclaudel) [inline doc/movies/extract_snippets/docutils.py:46]: This `None` should be the name of the current file

**Pull request merged**